src/ring_theory/witt_vector/truncated.lean

```diff
@@ -240,3 +238,221 @@ instance : comm_ring (truncated_witt_vector p n R) :=
+a truncated Witt vector of length `n`, for `n ≤ m`.
+-/
+def truncate {m : ℕ} (hm : n ≤ m) : truncated_witt_vector p m R →+* truncated_witt_vector p n R :=
+ring_hom.lift_of_surjective
+  (witt_vector.truncate m)
+  (witt_vector.truncate_surjective p m R)
+  (witt_vector.truncate n)
+  begin
+    intro x,
+    simp only [witt_vector.mem_ker_truncate],
+    intros h i hi,
+    exact h i (lt_of_lt_of_le hi hm)
+  end
+
+@[simp] lemma truncate_comp_witt_vector_truncate {m : ℕ} (hm : n ≤ m) :
+  (@truncate p _ n R _ m hm).comp (witt_vector.truncate m) = witt_vector.truncate n :=
+ring_hom.lift_of_surjective_comp _ _ _ _
+
+@[simp] lemma truncate_witt_vector_truncate {m : ℕ} (hm : n ≤ m) (x : 𝕎 R) :
+  truncate hm (witt_vector.truncate m x) = witt_vector.truncate n x :=
+ring_hom.lift_of_surjective_comp_apply _ _ _ _ _
+
+@[simp] lemma truncate_truncate {n₁ n₂ n₃ : ℕ} (h1 : n₁ ≤ n₂) (h2 : n₂ ≤ n₃)
+  (x : truncated_witt_vector p n₃ R) :
+  (truncate h1) (truncate h2 x) = truncate (h1.trans h2) x :=
+begin
+  obtain ⟨x, rfl⟩ := witt_vector.truncate_surjective p n₃ R x,
+  simp only [truncate_witt_vector_truncate],
+end
+
+@[simp] lemma truncate_comp {n₁ n₂ n₃ : ℕ} (h1 : n₁ ≤ n₂) (h2 : n₂ ≤ n₃) :
+  (@truncate p _ _ R _ _ h1).comp (truncate h2) = truncate (h1.trans h2) :=
+begin
+  ext1 x, simp only [truncate_truncate, function.comp_app, ring_hom.coe_comp]
+end
+
+lemma truncate_surjective {m : ℕ} (hm : n ≤ m) : surjective (@truncate p _ _ R _ _ hm) :=
+begin
+  intro x,
+  obtain ⟨x, rfl⟩ := witt_vector.truncate_surjective p _ R x,
+  exact ⟨witt_vector.truncate _ x, truncate_witt_vector_truncate _ _⟩
+end
+
+@[simp] lemma coeff_truncate {m : ℕ} (hm : n ≤ m) (i : fin n) (x : truncated_witt_vector p m R) :
+  (truncate hm x).coeff i = x.coeff (fin.cast_le hm i) :=
+begin
+  obtain ⟨y, rfl⟩ := witt_vector.truncate_surjective p _ _ x,
+  simp only [truncate_witt_vector_truncate, witt_vector.coeff_truncate, fin.coe_cast_le],
+end
+
+section fintype
+omit hp
+
+instance {R : Type*} [fintype R] : fintype (truncated_witt_vector p n R) := pi.fintype
+
+variables (p n R)
+
+lemma card {R : Type*} [fintype R] :
+  fintype.card (truncated_witt_vector p n R) = fintype.card R ^ n :=
+by simp only [truncated_witt_vector, fintype.card_fin, fintype.card_fun]
+
+end fintype
+
+lemma infi_ker_truncate : (⨅ i : ℕ, (@witt_vector.truncate p _ i R _).ker) = ⊥ :=
+begin
+  rw [submodule.eq_bot_iff],
+  intros x hx,
+  ext,
+  simp only [witt_vector.mem_ker_truncate, ideal.mem_infi, witt_vector.zero_coeff] at hx ⊢,
+  exact hx _ _ (nat.lt_succ_self _)
+end
+
+end truncated_witt_vector
+
+namespace witt_vector
+open truncated_witt_vector (hiding truncate coeff)
+
+section lift
+
+variable [comm_ring R]
+variables {S : Type*} [comm_ring S]
+variable (f : Π k : ℕ, S →+* truncated_witt_vector p k R)
+variable f_compat : ∀ (k₁ k₂ : ℕ) (hk : k₁ ≤ k₂),
+           (truncated_witt_vector.truncate hk).comp (f k₂) = f k₁
```

Comment:
Would it be reasonable to require f_compat only for a subset of $$\mathbb{N}$$ covering the whole $$\mathbb{N}$$? I can imagine circumstances where one is explicitely given an exaustive family of morphisms which is indexed in a slightly different manner from the numbering of the truncated Witt vectors. Still, this gives rise to a map to the whole Witt ring.

Reply:
Thanks for the suggestion. I played around a bit with this idea, but I didn't find a nice and general statement. So I think it's best to write the glue once it is needed. Because then we know exactly what we want. It shouldn't be hard to write it in terms of the existing `lift` at that point.

Reply:
I see, it definitely makes sense.